internal/service/lakeformation/status.go

```diff
@@ -6,42 +6,41 @@ package lakeformation
 import (
 	"context"
 	"fmt"
+	"reflect"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/lakeformation"
-	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/lakeformation"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/lakeformation/types"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
 )
 
-func statusPermissions(ctx context.Context, conn *lakeformation.LakeFormation, input *lakeformation.ListPermissionsInput, tableType string, columnNames []*string, excludedColumnNames []*string, columnWildcard bool) retry.StateRefreshFunc {
+func statusPermissions(ctx context.Context, conn *lakeformation.Client, input *lakeformation.ListPermissionsInput, tableType string, columnNames []string, excludedColumnNames []string, columnWildcard bool) retry.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		var permissions []*lakeformation.PrincipalResourcePermissions
+		var permissions []awstypes.PrincipalResourcePermissions
 
-		err := conn.ListPermissionsPagesWithContext(ctx, input, func(resp *lakeformation.ListPermissionsOutput, lastPage bool) bool {
-			for _, permission := range resp.PrincipalResourcePermissions {
-				if permission == nil {
+		pages := lakeformation.NewListPermissionsPaginator(conn, input)
+		for pages.HasMorePages() {
+			page, err := pages.NextPage(ctx)
+
+			if errs.IsA[*awstypes.EntityNotFoundException](err) || errs.IsA[*awstypes.InvalidInputException](err) {
+				return nil, statusNotFound, err
+			}
```

Comment:
Actually, after writing up an issue I think this should change slightly to match the previous implementation:

```suggestion
			if errs.IsA[*awstypes.EntityNotFoundException](err) {
				return nil, statusNotFound, err
			}
			
			if errs.IsAContains[*awstypes.InvalidInputException](err, "Invalid principal") {
				return nil, statusIAMDelay, err
			}
```